actionpack/lib/action_controller/metal/instrumentation.rb

```diff
@@ -35,6 +35,9 @@ def process_action(*)
         payload[:response] = response
         payload[:status]   = response.status
         result
+      rescue => e
+        payload[:status] = ActionDispatch::ExceptionWrapper.status_code_for_exception(e.class.name)
+        raise e
```

Comment:
Two things:

  - Personal nitpick, please use `error` rather than `e`, single letter variable are an eyesore outside of inline blocks.
  - Reraise the exception with `raise` without argument to not reset the backtrace.

Reply:
Rebased on main and pushed